documentation/projects/proposals/trust_and_safety/content_report_moderation/20230517-implementation_plan_mature_copy_change.md

```diff
@@ -0,0 +1,235 @@
+when possible, but without the need for database migrations (i.e., using
+[Django's built-in tools for specifying different underlying names for fields and tables](https://docs.djangoproject.com/en/4.2/ref/models/options/#table-names)).
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+All references to `mature` in code and copy will be updated to `sensitive`,
+except where `mature` is
+[part of a data model](https://github.com/WordPress/openverse/blob/0f90f84d4bf856297356a918e4cc4fe42d07ffc2/ingestion_server/ingestion_server/elasticsearch_models.py#L118)
+or makes the most sense to keep (e.g. when interacting with certain provider
+aspects).
+
+## Outlined Steps
+
+<!-- Describe the implementation step necessary for completion. -->
+
+The work is broken up into two services (API and frontend) and those services
+are broken into two aspects (code and copy).
+
+### API
+
+#### Code
+
+The following models will need to be updated. Since we want to avoid a migration
+with this work, they will need to
+[explicitly reference the old table names using `Meta::db_table`](https://docs.djangoproject.com/en/4.2/ref/models/options/#table-names).
+
+- [`AbstractMatureMedia`](https://github.com/WordPress/openverse/blob/2041c5df1e9d5d1f9f37e7c177f2e70f61ea5dba/api/api/models/media.py#L334-L333)
+- [`MatureImage`](https://github.com/WordPress/openverse/blob/2041c5df1e9d5d1f9f37e7c177f2e70f61ea5dba/api/api/models/image.py#L83)
+- [`MatureAudio`](https://github.com/WordPress/openverse/blob/2041c5df1e9d5d1f9f37e7c177f2e70f61ea5dba/api/api/models/audio.py#L263)
+
+The
+[`mature` reason](https://github.com/WordPress/openverse/blob/2041c5df1e9d5d1f9f37e7c177f2e70f61ea5dba/api/api/models/media.py#L22-L21)
+(and
+[`mature_filtered`](https://github.com/WordPress/openverse/blob/2041c5df1e9d5d1f9f37e7c177f2e70f61ea5dba/api/api/models/media.py#L18-L17)
+value) in `AbstractMediaReport` definition will also need to be changed to
+`sensitive` and `sensitive_filtered` respectively. In order to remain backwards
+compatible with the frontend, incoming references on reports for `mature` will
+need to be mapped to the new value.
+
+Per the
+[detecting sensitive textual content project plan](https://docs.openverse.org/projects/proposals/trust_and_safety/detecting_sensitive_textual_content/20230309-project_proposal_detecting_sensitive_textual_content.html#designation-of-results-with-sensitive-terms),
+the `mature` field of the media serializer will remain, and will eventually have
+a `sensitivity` field added to it to encapsulate specific information about the
+reason behind a media being marked sensitive. Those changes are not under the
+purview of this IP.
+
+The Elasticsearch model property
+[`mature`](https://github.com/WordPress/openverse/blob/2041c5df1e9d5d1f9f37e7c177f2e70f61ea5dba/ingestion_server/ingestion_server/elasticsearch_models.py#L118)
+will also not be affected.
```

Comment:
Will the ES model property also change later in conjunction with the field addition to the serializer?

Reply:
I'm not sure! I think `sensitive` is actually something we might end up deriving from a few different places in Elasticsearch, and might not be something we store on the indexed records directly. @sarayourfriend did you have a different impression?

Reply:
`mature` is already a derived property. It theoretically can encode if a provider supplies the fact but also gets flipped to true for confirmed user reports. If we can update it to sensitive that would be ideal. It again would take more careful planning and potentially several PRs to do it in a backwards compatible way (similar to a database column rename, really).

Reply:
Although, even more ideally would be to rename it to `has_confirmed_sensitivity_report` and then when we ingest sensitivity from providers in the future to use yet another field for that. If we can make the name more explicit and prevent overloading the meaning in the future, that would be 100% better than the current ambiguous name.

Reply:
That makes sense! It seems like it might be outside the scope of this IP though, so I'll leave this as-is for now unless there are any objections.

Reply:
Totally, I did not think it was something we would handle in this. Totally fine by me to continue excluding it from the scope.